### PR TITLE
feat: Update KBC_DATA_TYPE_SUPPORT behavior

### PR DESCRIPTION
**Pull request by jachym-tousek-keboola**
Jira: https://keboola.atlassian.net/browse/PSGO-650

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,28 @@
 - [CLI](./docs/cli.md)
 - [Processor](./docs/processor.md)
 
+## Development
```
Comment (jachym-tousek-keboola) [inline README.md:22]: Development was previously in the Processor block for some reason so I accidentally moved it too in my previous PR. Moving it back now.

File: docs/processor.md [hunk 2]
```diff
@@ -24,8 +24,9 @@ It supports optional parameters:
 
 It automatically uses the `KBC_DATA_TYPE_SUPPORT` env variable to detect whether to add `columns` or `schema` field to the manifest when both are missing.
 
-- If `KBC_DATA_TYPE_SUPPORT` has value `hint` or is missing then it adds `columns`.
-- If `KBC_DATA_TYPE_SUPPORT` has value `authoritative` then it adds `schema`.
+- If `KBC_DATA_TYPE_SUPPORT` has value `none` or is missing then it adds `columns`.
+- If `KBC_DATA_TYPE_SUPPORT` has value `hint` then it adds `schema` without data types.
+- If `KBC_DATA_TYPE_SUPPORT` has value `authoritative` then it adds `schema` with data types.
```
Comment (jachym-tousek-keboola) [inline docs/processor.md:29]: Updated according to the new specification.

File: internal/pkg/processor/config/config_test.go [hunk 3]
```diff
@@ -31,7 +32,7 @@ func TestConfig(t *testing.T) {
 		assert.NoError(t, os.WriteFile(configPath, []byte(testData.input), 0o0644))
 
 		// Test
-		conf, err := LoadConfig(configPath, "hint")
+		conf, err := LoadConfig(configPath, testData.dataTypeSupport)
```
Comment (jachym-tousek-keboola) [inline internal/pkg/processor/config/config_test.go:35]: Previously this test was always using the "hint" mode so "authoritative" was not covered. Added the missing tests now.

File: internal/pkg/slicer/config/config.go [hunk 4]
```diff
@@ -61,7 +61,8 @@ type Config struct {
 	BufferSize datasize.ByteSize `json:"bufferSize" mapstructure:"buffer-size" validate:"min=32768"`
 
 	// DataTypeMode changes how the table schema is added into manifest when missing.
-	DataTypeMode DataTypeMode `json:"-" mapstructure:"data-type-mode" validate:"required,oneof=columns schema"`
+	DataTypeMode      DataTypeMode `json:"-" mapstructure:"data-type-mode" validate:"required,oneof=columns schema"`
+	OptionalDataTypes bool         `json:"-"`
```
Comment (jachym-tousek-keboola) [inline internal/pkg/slicer/config/config.go]: CLI doesn't support the new mode, only processor does. So I had to solve it as separate field with default to how CLI works.

File: test/processor/data-type-hint/expected/data/out/tables/table.csv.manifest [hunk 5]
```diff
@@ -1,6 +1,10 @@
...
-        "id",
-        "val"
+    "schema": [
+        {
+            "name": "id"
+        },
+        {
+            "name": "val"
+        }
     ]
```
Comment (jachym-tousek-keboola) [inline test/processor/data-type-hint/expected/data/out/tables/table.csv.manifest:9]: hint mode was changed to new schema
the new test `data-type-none` is exact copy of what `data-type-hint` used to be (just with changed env value)

**Review by Matovidlo — APPROVED** [review]
LGTM

File: internal/pkg/manifest/manifest.go [hunk 6]
```diff
@@ -133,16 +133,18 @@ func (m *Manifest) SetColumns(columns []string) {
 	m.modified = true
 }
 
-func (m *Manifest) SetSchemaColumns(columns []string) {
+func (m *Manifest) SetSchemaColumns(columns []string, optionalDataTypes bool) {
```
Comment (michaljurecko) [inline internal/pkg/manifest/manifest.go]: `optionalDataTypes` - I don't know if it's the best naming, it wasn't clear to me whether it turns the types on or off.

Maybe `addTypes bool` ? or `withTypes bool`

**Review by michaljurecko — APPROVED** [review]
LGTM

File: internal/pkg/manifest/manifest.go [hunk 7]
```diff
@@ -133,12 +133,12 @@ func (m *Manifest) SetColumns(columns []string) {
 	m.modified = true
 }
 
-func (m *Manifest) SetSchemaColumns(columns []string, optionalDataTypes bool) {
+func (m *Manifest) SetSchemaColumns(columns []string, addTypes bool) {
 	schema := []*orderedmap.OrderedMap{}
```
Comment (michaljurecko) [inline internal/pkg/manifest/manifest.go]: :+1:

**Pull request merged**